### DISCUSSION TITLE
Alpha 0.4.2

### DISCUSSION THREAD

**Discussion by valeriewong052** [spacedriveapp/spacedrive#2672] [Announcements]
Our final patch before Beta! This release brings UI improvements, package updates, platform migrations, and bug fixes to improve stability. More Chinese translations have also been added.

## 🔧 Core Updates
* Spacedrive now runs on a more stable version of Tauri v2 - by [@Rocky43007](https://github.com/Rocky43007) in [#2654](https://github.com/spacedriveapp/spacedrive/pull/2654)

### 🖥️ Desktop
* Dimmed the text color on dark mode and increased the text size for keybinds - by [@PineappleRind](https://github.com/PineappleRind) in [#2643](https://github.com/spacedriveapp/spacedrive/pull/2643)

(Before/After)
<img width="300" alt="before" src="https://github.com/user-attachments/assets/a5cb5184-b028-4a30-a370-f6781c43d7c9"></img><img width="300" alt="after" src="https://github.com/user-attachments/assets/d8348c27-bde9-47d4-b131-4c30274d6fbe"></img>

* Job Manager now shows progress when copying files - by [@matheus-consoli](https://github.com/matheus-consoli) in [#2538](https://github.com/spacedriveapp/spacedrive/pull/2538)

<img width="400" alt="Screenshot for report progress when copying files feature" src="https://github.com/user-attachments/assets/d76c0916-9a99-4e0a-aa59-9414eefd67d8"></img>

### 🖥️ Desktop
* Fixed a bug where pressing <kbd>Esc</kbd> would exit both the Quick Preview and app search simultaneously - by [@myung03](https://github.com/myung03) in [#2646](https://github.com/spacedriveapp/spacedrive/pull/2646)
* Fixed a bug causing "zombie" jobs to appear in the job manager - by [@myung03](https://github.com/myung03) in [#2653](https://github.com/spacedriveapp/spacedrive/pull/2653)
<img width="360" alt="Screenshot 2024-07-29 at 11 08 55 AM" src="https://github.com/user-attachments/assets/22947389-f3af-4bd9-a4d6-2c688c4fff1c"></img>
* Fixed a [bug](https://github.com/ajrcarey/pdfium-render/pull/155) with PDF thumbnails failing to render - by [@HeavenVolkoff](https://github.com/HeavenVolkoff) in [#2666](https://github.com/spacedriveapp/spacedrive/pull/2666)
* Fixed a [bug](https://github.com/spacedriveapp/spacedrive/issues/2647) with opening Spacedrive on Debian Linux - by [@HeavenVolkoff](https://github.com/HeavenVolkoff) in [#2648](https://github.com/spacedriveapp/spacedrive/pull/2648)
* [Fixed a bug preventing Spacedrive from launching](https://github.com/user-attachments/assets/61fc6573-daba-4906-bebb-2004e59f1c4c) - by [@HeavenVolkoff](https://github.com/HeavenVolkoff) in [#2670](https://github.com/spacedriveapp/spacedrive/pull/2670)

## 🛠️ General/Development Updates
* Added additional Chinese translations - by [@Lkhsss](https://github.com/Lkhsss) in [#2656](https://github.com/spacedriveapp/spacedrive/pull/2656)
* Updated the `Vite` dependency to 5.2.0 - by [@Rocky43007](https://github.com/Rocky43007) in [#2645](https://github.com/spacedriveapp/spacedrive/pull/2645)
* Removed a number of unused code modules - by [@fogodev](https://github.com/fogodev) ([#2660](https://github.com/spacedriveapp/spacedrive/pull/2660)) and [@HeavenVolkoff](https://github.com/HeavenVolkoff) ([#2662](https://github.com/spacedriveapp/spacedrive/pull/2662))
* Updated various build/release tools and server scripts - by [@HeavenVolkoff](https://github.com/HeavenVolkoff) in [#2642](https://github.com/spacedriveapp/spacedrive/pull/2642), [#2663](https://github.com/spacedriveapp/spacedrive/pull/2663), [#2668](https://github.com/spacedriveapp/spacedrive/pull/2668)
* Updated a number of Rust, Prisma, and Windows dependencies - by [@HeavenVolkoff](https://github.com/HeavenVolkoff) in [#2662](https://github.com/spacedriveapp/spacedrive/pull/2662), [#2663](https://github.com/spacedriveapp/spacedrive/pull/2663)
* Miscellaneous updates to documentation, code style, and version references - by [@iLynxcat](https://github.com/iLynxcat) ([#2639](https://github.com/spacedriveapp/spacedrive/pull/2639)) and [@HeavenVolkoff](https://github.com/HeavenVolkoff) ([#2670](https://github.com/spacedriveapp/spacedrive/pull/2670), [#2667](https://github.com/spacedriveapp/spacedrive/pull/2667))

### 📱 Mobile (soon)<sup>tm</sup>
* Updated the `Expo SDK` dependency to 51.0.28 - by [@Rocky43007](https://github.com/Rocky43007) in [#2665](https://github.com/spacedriveapp/spacedrive/pull/2665)

## 🆕 New Contributors
* [@Lkhsss](https://github.com/Lkhsss) made their first contribution in [#2656](https://github.com/spacedriveapp/spacedrive/pull/2656)

## 🚀 What's Next?

We're thrilled to announce that the next planned update **(version 0.5)** for Spacedrive will mark a significant milestone as we transition from the Alpha stage to a full Beta release. This major update will introduce the highly anticipated **cloud sync** and **P2P (peer-to-peer) capabilities**, allowing seamless file synchronization across devices and collaboration.

In addition, we're excited to unveil the **Alpha release of our mobile app**! Users can also look forward to a variety of smaller features and UI updates to explore across the board.

Thank you for using Spacedrive — we hope you're all as excited as we are for this upcoming release!

**Full Changelog**: [https://github.com/spacedriveapp/spacedrive/compare/0.4.1...0.4.2](https://github.com/spacedriveapp/spacedrive/compare/0.4.1...0.4.2)

<hr/>

<em>Note: This is a re-release of Alpha 0.4.2 due to an expected error with an earlier release. If you previously downloaded the 0.4.2 release (and your app crashes/fails to launch): 

1. Uninstall Spacedrive from your device
2. Install a clean version of **[Spacedrive 0.4.1](https://github.com/spacedriveapp/spacedrive/releases/tag/0.4.1)** 
3. Update Spacedrive using the app update feature
<img width="242" alt="Screenshot 2024-08-17 at 7 11 46 PM" src="https://github.com/user-attachments/assets/80623f09-bd2b-43b1-83d0-7bc2ac96e7f2"></img>
 </em>

<hr /><em>This discussion was created from the release <a href='https://github.com/spacedriveapp/spacedrive/releases/tag/0.4.2'>Alpha 0.4.2</a>.</em>

**Comment by mon-jai** [comment]
Congrats on the new release! Looking forward to having a nightly APK available for the Android app.

**Reply by arxari-archive** [reply]
Same!

**Comment by arxari-archive** [comment]
Just found this thanks to It's Foss, this file manager is honestly next level.

**Comment by arran4** [comment]
Any chance of a standalone linux executable, AppImage or flatpak?

**Reply by valeriewong052** [reply]
We’re currently working on a flatpak release package, but afaik we had to discontinue AppImage due to some compatibility issues! No current plans for a standalone executable due to filesize concerns (could be revisited at some point in the future though!)

**Reply by arran4** [reply]
Personally not that concerned about a single file b/c I use things like llama file which are gb in size:

https://huggingface.co/Mozilla/whisperfile

**Comment by sscorpio91** [comment]
I am loving what it can and will do. I'm hopeful connecting nodes will come soon. I am experiencing one glitch in particular, but I am a special case, perhaps. When I open a directory that is empty it will immediately show me that. If there are files, I have to open a modifier, ie. open the inspector (Ctrl + i) then return to default for the files to appear. I say this may be special because I am running the .deb on Void Linux. If I am the only one experiencing this, it's clearly an issue with the conversion to xbps. 
Otherwise, all-in-all, great work! Keep it up! I look forward to the next phase, and ultimately the Release. I will be adding this to my Arch as well later today. But that will be run through a Debian container, rather than performing a conversion.

**Comment by saner99** [comment]
adding AppImage to future releases would be very useful

it should not cause any inconvenience and will be much easier for many Linux users before installing Spacedrive will be possible using package managers

so its looks better for client apps than just .deb (i use Arch, and often prefer AppImage)

**Reply by sscorpio91** [reply]
I agree, however, The downside of appimage is its lack of musl support. Probably the other way round. Either way, AIs do not work on musl. I'm sure it's niche. Alpine, some Void, Chimera, and I'm sure a few others.
Is there really a universal and agnostic way? AI depends on glibc, flatpak, I believe, depends on elogind, snaps on systemd.
This is not against you, you just happened to have posted. - Like I said, I do like the thought of AppImage, I also see the limitation.
